src/dma.rs

```diff
@@ -0,0 +1,1999 @@
+                    dma.$cr.modify(|_, w| unsafe { w.dir().bits(direction.bits()) });
+                }
+
+                /// Convenience method to configure the 4 common interrupts for the DMA stream
+                pub fn set_interrupts(dma: &mut $DMA, transfer_complete: bool, half_transfer: bool, transfer_error: bool, direct_mode_error: bool) {
+                    dma.$cr.modify(|_, w| w
+                        .tcie().bit(transfer_complete)
+                        .htie().bit(half_transfer)
+                        .teie().bit(transfer_error)
+                        .dmeie().bit(direct_mode_error)
+                    );
+                }
+
+                /// Enable/disable the transfer complete interrupt (tcie) of the DMA stream
+                pub fn set_transfer_complete_interrupt(dma: &mut $DMA, transfer_complete_interrupt: bool) {
+                    dma.$cr.modify(|_, w| w.tcie().bit(transfer_complete_interrupt));
+                }
+
+                /// Enable/disable the half transfer interrupt (htie) of the DMA stream
+                pub fn set_half_transfer_interrupt(dma: &mut $DMA, half_transfer_interrupt: bool) {
+                    dma.$cr.modify(|_, w| w.htie().bit(half_transfer_interrupt));
+                }
+
+                /// Enable/disable the transfer error interrupt (teie) of the DMA stream
+                pub fn set_transfer_error_interrupt(dma: &mut $DMA, transfer_error_interrupt: bool) {
+                    dma.$cr.modify(|_, w| w.teie().bit(transfer_error_interrupt));
+                }
+
+                /// Enable/disable the direct mode error interrupt (dmeie) of the DMA stream
+                pub fn set_direct_mode_error_interrupt(dma: &mut $DMA, direct_mode_error_interrupt: bool) {
+                    dma.$cr.modify(|_, w| w.dmeie().bit(direct_mode_error_interrupt));
+                }
+
+                /// Enable/disable the fifo error interrupt (feie) of the DMA stream
+                pub fn set_fifo_error_interrupt(dma: &mut $DMA, fifo_error_interrupt: bool) {
+                    dma.$fcr.modify(|_, w| w.feie().bit(fifo_error_interrupt));
+                }
+
+                /// Enable/disable the double buffer (dbm) of the DMA stream
+                pub fn set_double_buffer(dma: &mut $DMA, double_buffer: bool) {
+                    dma.$cr.modify(|_, w| w.dbm().bit(double_buffer));
+                }
+
+                /// Set the fifo threshold (fcr.fth) of the DMA stream
+                pub fn set_fifo_threshold(dma: &mut $DMA, fifo_threshold: config::FifoThreshold) {
+                    dma.$fcr.modify(|_, w| w.fth().bits(fifo_threshold.bits()));
+                }
+
+                /// Enable/disable the fifo (dmdis) of the DMA stream
+                pub fn set_fifo_enable(dma: &mut $DMA, fifo_enable: bool) {
+                    //Register is actually direct mode disable rather than fifo enable
+                    dma.$fcr.modify(|_, w| w.dmdis().bit(!fifo_enable));
```

Comment:
Just passing through, I've adapted your PR for a F2 board I am using and its working well, Thanks!

Only one issue I had was in `set_fifo_enable`. This should be just `fifo_enable`, inverting does the opposite.